tests/src/python/test_qgsgeometry.py

```diff
@@ -2334,6 +2334,18 @@ def testReshape(self):
         wkt = g.asWkt()
         assert compareWkt(expWkt, wkt), "testReshape failed: mismatch Expected:\n%s\nGot:\n%s\n" % (expWkt, wkt)
 
+        # Test reshape a polygon with a line starting or ending at the polygon's first vertex, no change expexted
+        g = QgsGeometry.fromWkt('Polygon ((0 0, 1 0, 1 1, 0 1, 0 0))')
+        expWkt = g.asWkt()
+        g.reshapeGeometry(QgsLineString([QgsPoint(0, 0), QgsPoint(-1, -1)]))
```

Comment:
```suggestion
        self.assertEqual(g.reshapeGeometry(QgsLineString([QgsPoint(0, 0), QgsPoint(-1, -1)])), QgsGeometry.Success)
```

It would be good to check the operation result too.

Reply:
That's true, but the expected result is not a `QgsGeometry.Success` but rather an `QgsGeometry.OperationResult.NothingChanged`, so until that is fixed the operation result is better not checked! I'll add it to the second test that should be a success.